src/Utils/ChangeNotifier.js

```diff
@@ -0,0 +1,74 @@
+import config from './../config/config.js';
+
+class ChangeNotifier  {
+	
+	constructor() {
+		if(!config.useBackendMock) {
+			this.webSocket = new WebSocket(config.backendWebSocketURL);
+			this.webSocket.onerror = this.handleError;
+			this.webSocket.onopen = this.handleNewConnection;
+			this.webSocket.onclose = this.handleConnectionClose;
+			this.webSocket.onmessage = this.handleMessage.bind(this);
+		}
+		this.subscribers = [];
+		this.handleMessage = this.handleMessage.bind(this);
+		this.register = this.register.bind(this);
+		this.unregisterAll = this.unregisterAll.bind(this);
+	}
+
+	static handleNotification(subscription, notification) {
+		ChangeNotifier.handleNotificationForGridView(subscription, notification);
+		ChangeNotifier.handleNotificationForDetailView(subscription, notification);
+    }
+
+    static handleNotificationForGridView(subscription, notification) {
+        if (notification.ArtifactType === "Entity" && subscription.artifactId === notification.ArtifactId) {
+			subscription.notificationCallback();
+        }
+    }
+
+    static handleNotificationForDetailView(subscription, notification) {
+		if (
+			(notification.ArtifactType === "Entity" && subscription.artifactId === notification.ArtifactId)
+			||
+			(notification.ArtifactType === "Event" && subscription.artifactId === parseInt(notification.EntityId, 10))
+		) {
+			subscription.notificationCallback(notification.EventTypeId);
+		}
+	}
+
+	handleNewConnection() {
+		console.log("[WEBSOCKET] Established a new connection! Connection is open.");
+	}
+	
+	handleError() {
+		console.log("[WEBSOCKET] An error occured! Connection could not be established.");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS2228)